Thermomachines code cleaning + some QoL

Menshin opened the pull request:
## About the PR

Cleaned most of the thermomachine code and added some Quality of Life changes, including :

* removed some unnecessary old code (the UI and sprite appeareance shouldn't update every tick when the machine isn't enabled anymore) and used wrappers when possible.
* visual feedback on the maintenance panel state:
![S9ww5IDhBo](https://user-images.githubusercontent.com/7117411/226993027-b623722c-4695-4dea-a08d-57ef348b1534.gif)
* active power consumption handling : the machine should now only use APC power when actually running (i.e heater/freezing gas).  **That means that some maps will have to be updated because some parts of atmos are not within powered range of a LV cable by default**

![o23va9mGBa](https://user-images.githubusercontent.com/7117411/226995677-52fef765-e41d-4368-9676-8cfa2511e339.gif)

**Thermomachines are now off by default** , but i've made Enabled prototype variants (suffixed) for mappers that want to use switched on thermomachines on spawn instead.
Unachoring/anchoring handled.
* Detailed examine that show the temperature set if close enough.
![image](https://user-images.githubusercontent.com/7117411/231821528-9769839e-05da-41bf-b123-62b8b9510859.png)


- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**
<!--
Here you can fill out a changelog that will automatically be added to the game when your PR is merged.

Only put changes that are visible and important to the player on the changelog.

Don't consider the entry type suffix (e.g. add) to be "part" of the sentence:
bad: - add: a new tool for engineers
good: - add: added a new tool for engineers

Putting a name after the :cl: symbol will change the name that shows in the changelog (otherwise it takes your GitHub username)
Like so: :cl: PJB
-->

:cl:
- fix: Heaters/freezers aren't constantly updating UI/sprite when turned off anymore. It should now be possible to set temperature in such cases.
- add: Heaters/freezers now uses power only when actually running. That also means, they have to be powered to run now.
- add: Visual feedback for when heaters/freezers maintenance panel is open
- add: Heaters/freezers set temperature should now appear on detailed examined (i.e when close enough)

Menshin commented:
Drafted until the weird comp is changed on spawn CI fail is fixed.

metalgearsloth commented on Resources/Textures/Structures/Piping/Atmospherics/thermomachine.rsi/meta.json:
```diff
@@ -1 +1,56 @@
+			"directions":1
+		},
+		{
+			"name":"freezerOn",
+			"directions":1,
+			"delays":[ [ 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1] ]
+		},
+		{
+			"name":"freezerPanelOn",
+			"directions":1,
+			"delays":[ [ 0.1, 0.1] ]
+		},
+		{
+			"name":"heaterOff",
+			"directions":1
+		},
+		{
+			"name":"heaterPanelOpen",
+			"directions":1
+		},
+				{
+			"name":"heaterPanelClose",
+			"directions":1
+		},
+		{
+			"name":"heaterOn",
+			"directions":1,
+			"delays":[ [ 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1] ]
+		},
+		{
+			"name":"heaterPanelOn",
+			"directions":1,
+			"delays":[ [ 0.1, 0.1] ]
+		},
+		{
+			"name":"pipe",
+			"directions":4
+		}
+	]
+}
```
Formatting needs fixing, please don't use tabs.

metalgearsloth commented on Content.Server/Atmos/Piping/Unary/Components/GasThermoMachineComponent.cs:
```diff
@@ -13,7 +13,13 @@ public sealed class GasThermoMachineComponent : Component
-        public bool Enabled = true;
+        public bool Enabled = false;
+
+        /// <summary>
+        ///      Whether the power switch is on AND the machine has enough power
+        /// </summary>
+        [ViewVariables]
+        public bool IsRunning = false;
```
This doesn't look necessary.

metalgearsloth commented on Content.Server/Atmos/Piping/Unary/EntitySystems/GasThermoMachineSystem.cs:
```diff
@@ -5,42 +5,58 @@
+
         private void OnThermoMachineUpdated(EntityUid uid, GasThermoMachineComponent thermoMachine, AtmosDeviceUpdateEvent args)
         {
-            if (!thermoMachine.Enabled
-                || !EntityManager.TryGetComponent(uid, out NodeContainerComponent? nodeContainer)
+
+            if (!thermoMachine.IsRunning
```
Can't this just use enabled + IsPowered still?

metalgearsloth commented on Content.Client/Atmos/Visualizers/GasThermoMachineVisualsSystem.cs:
```diff
@@ -0,0 +1,47 @@
+using Content.Shared.Wires;
+using Content.Client.Wires.Visualizers;
+
+namespace Content.Client.Atmos.Visualizers
+{   /// <summary>
+    /// Controls client-side visuals for gas thermomachines (freezers/heaters).
+    /// </summary>
+    public sealed class GasThermoMachineSystem : VisualizerSystem<GasThermoMachineVisualsComponent>
+    {
+        protected override void OnAppearanceChange(EntityUid uid, GasThermoMachineVisualsComponent visuals, ref AppearanceChangeEvent args)
+        {
+            if (args.Sprite == null)
+                return;
+
+
+            args.AppearanceData.TryGetValue(ThermoMachineVisuals.Running, out var isRunningObject);
```
Formatting needs fixing.

Also this should just be using the component rather than an entirely separate visuals component.

Menshin replied:
Yeah, realized that if i scrapped animations the generic visualizer is enough. I'm on it.

Menshin commented:
Since there's no longer custom animations on this, removed the visualizer and delegated appearance handling to the corresponding components (power and wires).
Fixed some formatting as requested.
Added some minor visual feedback on the Enabled button in the UI.

Partmedia commented:
Is there a lighter blue that would be more visible on a black background for the "the freezer thermostat is set to X" text?

Menshin commented:
Light blue (used for pumps)/ Light Sky Blue / Deep Sky Blue
![temp1](https://user-images.githubusercontent.com/7117411/231087015-4784a251-2372-4ce2-8340-12f2d7161ecc.PNG)
![Temp2](https://user-images.githubusercontent.com/7117411/231087024-1896c01e-3c5b-4079-8dc1-79ea8f9b4fae.PNG)
![Temp3](https://user-images.githubusercontent.com/7117411/231087035-b4664c94-a4ed-4df5-89a7-0049700456d5.PNG)

metalgearsloth commented on Content.Server/Atmos/Piping/Unary/EntitySystems/GasThermoMachineSystem.cs:
```diff
@@ -49,82 +50,90 @@ private void OnThermoMachineUpdated(EntityUid uid, GasThermoMachineComponent the
         }
 
-        private void OnThermoMachineLeaveAtmosphere(EntityUid uid, GasThermoMachineComponent component, AtmosDeviceDisabledEvent args)
+        private void OnGasThermoRefreshParts(EntityUid uid, GasThermoMachineComponent thermoMachine, RefreshPartsEvent args)
         {
-            _appearance.SetData(uid, ThermoMachineVisuals.Enabled, false);
```
Looks like this got removed?

I'm not familiar enough with atmos code to comment on some of these changes and not sure if stuff breaks.

Menshin replied:
The original code was :

```c#
        private void OnThermoMachineLeaveAtmosphere(EntityUid uid, GasThermoMachineComponent component, AtmosDeviceDisabledEvent args)
        {
            _appearance.SetData(uid, ThermoMachineVisuals.Enabled, false);

            DirtyUI(uid, component);
        }
```

So basically just appearance/UI updating (that is now  handled by the `ApcPowerReceiver` component).

> I'm not familiar enough with atmos code to comment on some of these changes and not sure if stuff breaks.

I've not touched any of the actual atmos logic (located in `OnThermoMachineUpdated`) and tested that at the best of my abilities (i.e in local), but i fully understand your stance (i don't know if or who i should ping though).

Thanks for the re-review anyway.
